app/models/server/models/LivechatVisitors.js

```diff
@@ -57,6 +57,12 @@ export class LivechatVisitors extends Base {
 		return this.find(query);
 	}
 
+	findByUsername(username, options) {
+		const query = { username };
+
+		return this.find(query, options);
```

Comment:
Why not returning just one Livechat visitor?
```suggestion
		return this.findOne(query, options);
```

Reply:
Yes, Let me update this

Reply:
@renatobecker if we use `findOne` here, the code will break. Then `getFullUserData(...).observeChanges` won't be a function.